src/main/java/io/fusionauth/http/server/io/HTTPInputStream.java

```diff
@@ -133,7 +137,7 @@ public int read(byte[] buffer, int offset, int length) throws IOException {
         bodyBytes = null;
       }
     } else {
-      read = delegate.read(buffer);
+      read = delegate.read(buffer, offset, length);
```

Comment:
This was the primary bug identified that was causing runtime issues on the new virtual thread version of `java-http`.